cosmwasm/packages/wasmi-runtime/src/wasm/errors.rs

```diff
@@ -1,45 +1,27 @@
-#![allow(unused)]
-
 use derive_more::Display;
-use enclave_ffi_types::EnclaveError;
+use enclave_ffi_types::{EnclaveError, UntrustedVmError};
 use wasmi::{Error as InterpreterError, HostError};
 
 #[derive(Debug, Display)]
 #[non_exhaustive]
 pub enum WasmEngineError {
-    FailedOcall,
+    #[display(fmt = "FailedOcall")]
+    FailedOcall(UntrustedVmError),
     OutOfGas,
+    Panic,
+
     EncryptionError,
     DecryptionError,
-    DbError(DbError),
+
     MemoryAllocationError,
     MemoryReadError,
     MemoryWriteError,
-    InputInvalid,
+
     InputEmpty,
```

Comment:
Looking at how this error variant is used, i think it should also return an error into the contract.

Reply:
https://github.com/enigmampc/SecretNetwork/issues/329